packages-exp/auth-exp/src/platform_browser/persistence/indexed_db.ts

```diff
@@ -193,14 +194,34 @@ class IndexedDBLocalPersistence implements Persistence {
     );
   }
 
-  private async initialize(): Promise<IDBDatabase> {
+  async _openDb(): Promise<IDBDatabase> {
     if (this.db) {
       return this.db;
     }
     this.db = await _openDatabase();
     return this.db;
   }
 
+  async _withRetries<T>(op: (db: IDBDatabase) => Promise<T>): Promise<T> {
+    let numAttempts = 0;
+
+    while (true) {
+      try {
+        const db = await this._openDb();
+        return await op(db);
+      } catch (e) {
+        if (numAttempts++ > IndexedDBLocalPersistence.TRANSACTION_RETRY_COUNT) {
+          throw e;
+        }
+        if (this.db) {
+          this.db.close();
+          this.db = undefined;
+        }
+        // TODO: consider adding exponential backoff
```

Comment:
Is it worth adding any sort of delay here?

Reply:
question for @schmidt-sebastian I think he mentioned they had this in Firestore but I couldn't find proof

Reply:
We retry immediately at this scope, but at the higher level we retry with backoff: https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/src/util/async_queue.ts#L361

Reply:
thanks, will consider adding that in a followup if this doesn't fix it